server/src/main/java/org/opensearch/cluster/routing/RoutingTable.java

```diff
@@ -540,9 +540,13 @@ public Builder addAsNew(IndexMetadata indexMetadata) {
 
         public Builder addAsRecovery(IndexMetadata indexMetadata) {
             if (indexMetadata.getState() == IndexMetadata.State.OPEN || isIndexVerifiedBeforeClosed(indexMetadata)) {
-                IndexRoutingTable.Builder indexRoutingBuilder = new IndexRoutingTable.Builder(indexMetadata.getIndex())
-                    .initializeAsRecovery(indexMetadata);
-                add(indexRoutingBuilder);
+                if (indexMetadata.getSettings().getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false) == false
+                    || indicesRouting.get(indexMetadata.getIndex().getName())
+                        .shardsMatchingPredicateCount(sr -> sr.recoverySource() instanceof RecoverySource.RemoteStoreRecoverySource) == 0) {
```

Comment:
for RemoteStoreRecoverySource, RoutingTable will not be populated. Now at what point RoutingTable will be populated. Is there an integ test which would pass now with this change and failing earlier?

Reply:
We populate the routing table when cluster state is updated in RemoteStoreRestoreService. so skipping should be fine. 

We did not have iteg tests which were failing earlier due to this reason.

Some old remote store recovery integ tests which were passing earlier have started to fail, those i need to handle

Reply:
> Some old remote store recovery integ tests which were passing earlier have started to fail, those i need to handle

why did those test fail?

Reply:
> We did not have iteg tests which were failing earlier due to this reason.

Correction. The new integ tests which we wrote for Remote Cluster state were already failing and after this change they are passing.

> Some old remote store recovery integ tests which were passing earlier have started to fail, those i need to handle

These were failing with the first version of the PR. now they are not failing as we have moved the check to a place where is less intrusive to existing flows